src/Forms/DateField.php

```diff
@@ -58,175 +33,320 @@
+     */
+    protected function renderWithClientView($callback)
+    {
+        $clientView = null;
+        if ($this->getShowCalendar()) {
+            $clientView = $this->getClientView();
+            $clientView->onBeforeRender();
+        }
+        $html = $callback();
+        if ($clientView) {
+            $html = $clientView->onAfterRender($html);
         }
-        $config = array_filter($config);
-        foreach ($config as $k => $v) {
-            $this->setAttribute('data-' . $k, $v);
+        return $html;
+    }
+
+    public function getAttributes()
+    {
+        $attributes = parent::getAttributes();
+
+        // Merge with client config
+        $config = $this->getClientConfig();
+        foreach ($config as $key => $value) {
+            $attributes["data-{$key}"] = $value;
+        }
+
+        return $attributes;
+    }
+
+    public function Field($properties = array())
+    {
+        if (!$this->getSeparateDMYFields()) {
+            return parent::Field($properties);
         }
 
         // Three separate fields for day, month and year
-        if ($this->getConfig('dmyfields')) {
-            // values
-            $valArr = ($this->valueObj) ? $this->valueObj->toArray() : null;
-
-            // fields
-            $fieldNames = Zend_Locale::getTranslationList('Field', $this->locale);
-            $fieldDay = NumericField::create($this->name . '[day]', false, ($valArr) ? $valArr['day'] : null)
-                ->addExtraClass('day')
-                ->setAttribute('placeholder', $this->getConfig('dmyplaceholders') ? $fieldNames['day'] : null)
-                ->setMaxLength(2);
-
-            $fieldMonth = NumericField::create($this->name . '[month]', false, ($valArr) ? $valArr['month'] : null)
-                ->addExtraClass('month')
-                ->setAttribute('placeholder', $this->getConfig('dmyplaceholders') ? $fieldNames['month'] : null)
-                ->setMaxLength(2);
-
-            $fieldYear = NumericField::create($this->name . '[year]', false, ($valArr) ? $valArr['year'] : null)
-                ->addExtraClass('year')
-                ->setAttribute('placeholder', $this->getConfig('dmyplaceholders') ? $fieldNames['year'] : null)
-                ->setMaxLength(4);
-
-            // order fields depending on format
-            $sep = $this->getConfig('dmyseparator');
-            $format = $this->getConfig('dateformat');
-            $fields = array();
-            $fields[stripos($format, 'd')] = $fieldDay->Field();
-            $fields[stripos($format, 'm')] = $fieldMonth->Field();
-            $fields[stripos($format, 'y')] = $fieldYear->Field();
-            ksort($fields);
-            $html = implode($sep, $fields);
-
-            // dmyfields doesn't work with showcalendar
-            $this->setConfig('showcalendar', false);
-        } // Default text input field
-        else {
-            $html = parent::Field();
+        $valArr = $this->iso8601ToArray($this->Value());
```

Comment:
This breaks when `locale` isn't `en_us`, since `Value()` uses `iso8601ToLocalised()`, but `iso8601ToArray()` expects ISO